app/models/study.rb

```diff
@@ -1250,9 +1250,11 @@ def all_cells
   # return an array of all single cell names in study, will check for master list of cells or concatenate all
   # cell lists from individual expression matrices
   def all_cells_array
-    arrays = DataArray.where(study_id: self.id, linear_data_type: 'Study', linear_data_id: self.id, name: 'All Cells').order_by(&:array_index)
-    if arrays.any?
-      arrays.pluck(:values).reduce(&:+)
+    if self.metadata_file.present? && self.metadata_file.parsed?
```

Comment:
Since we're optimizing, it seems we could shave off more time by:

```suggestion
    if self.metadata_file.parsed?
```

(`self.metadata_file.present? == false and self.metadata_file.parsed? == true` seems impossible.)

Reply:
That is correct, but if there is no metadata file present in the study, then `self.metadata_file.parsed?` results in a `NoMethodError`.  The `self.metadata_file.present?` call first is to simply avoid that.

Reply:
This seems like a good place for nil-safe function calling: `self.metadata_file&.parsed?`